app/editor/src/features/navbar/NavBar.tsx

```diff
@@ -7,8 +7,47 @@ import { Row } from 'tno-core/dist/components/flex/row';
  */
 export const NavBar: React.FC = () => {
   const [activeHover, setActiveHover] = useState<string>('');
+
+  let isHover = false;
```

Comment:
I don't think this will work.  React state would change these values randomly at any point.

Reply:
It seems working based on the testing here, could we have a quick chat about this? Thanks

Reply:
`  const [activeHover, setActiveHover] = useState('');
  const [hide, setHide] = useState(false);

  const ref = React.useRef<any>();

  React.useEffect(() => {
    document.addEventListener('click', handleClickOutside, false);
    return () => {
      document.removeEventListener('click', handleClickOutside, false);
    };
  });

  React.useEffect(() => {
    if (hide) {
      setActiveHover('');
      setHide(false);
    }
  }, [hide]);

  const onMouseOver = () => {
    setHide(false);
  };

  const onMouseLeave = () => {
    if (!!activeHover) {
      setTimeout(() => {
        if (hide) setHide(true);
      }, 2000);
    }
  };

  const handleClickOutside = (event: { target: any }) => {
    if (ref.current && !ref.current.contains(event.target)) {
      setHide(true);
    }
  };

  return (
    <div onMouseOver={onMouseOver} onMouseLeave={onMouseLeave} ref={ref}>`

Reply:
Updated, and it looks like a work around has been found for this. Thanks